### ISSUE TITLE
reflowAndAddClass() reflows even when there's no class to add

### ISSUE THREAD

**Issue by michaelkix** [reactjs/react-transition-group#346]
**Do you want to request a *feature* or report a *bug*?**

Micro-optimization or performance bug, not sure what to call it.

**What is the current behavior?**

reflowAndAddClass() reflows even when there's no class to add.

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via https://jsfiddle.net or similar (template: https://jsfiddle.net/m4mb9Lg3/4/).**

Define a CSSTransition with exit and exitActive classes, but no enter/enterActive classes.

**What is the expected behavior?**

As far as I understand, there's no need to reflow if there's no class to add.

I would suggest something like this:

```javascript
reflowAndAddClass(node, className) {
    // This is for to force a repaint,
    // which is necessary in order to transition styles when adding a class name.
    if (className) {
        /* eslint-disable no-unused-expressions */
        node && node.scrollTop;
        /* eslint-enable no-unused-expressions */
        addClass(node, className);
    }
}
```

If you wonder why I have empty class names in the first place, I can explain. I couldn't avoid it, possibly due to another bug or just quirky behavior.

**Comment by jquense** [comment]
sounds good, care to send a PR?

**Comment by silvenon** [comment]
Thanks! No, it's related to #335.

**Comment by silvenon** [comment]
Fixed in #349.
